File: cpp/n482_FormatKey.cpp
// https://leetcode-cn.com/problems/license-key-formatting/
#include <iostream>
#include <string>
#include <vector>
using namespace std;

class Solution{
public:
    string licenseKeyFormatting(string S, int K){
        for(size_t i = 0; i < S.size();){
            if (S[i] == '-') S.erase(i, 1);
            else {
                S[i] = toupper(S[i]);
                i += 1;
            }
        }
        int l = S.size() - K;
        while (l > 0){
            S.insert(l, 1, '-');
            l -= K;
        }
        return S;
    }
};

int main(){
    Solution solu;
    string S = "5F3Z-2e-9-w";
    int K = 4;
    cout << solu.licenseKeyFormatting(S, K) << endl;
}

File: python/n482_FormatKey.py
# https://leetcode-cn.com/problems/license-key-formatting/
class Solution:
    def licenseKeyFormatting(self, S: str, K: int) -> str:       
        S = S.replace("-", "")
        S = S.upper()
        lis = list(S)
        l = len(S) - K
        while l > 0:           
            lis.insert(l, '-')
            l = l - K
        S = ''.join(lis)
        return S
        
    def licenseKeyFormatting_a(self, S: str, K: int) -> str:       
        S = S.replace("-", "")
        S = S.upper()
        l = len(S) - K
        while l > 0:           
            S = S[0:l] + "-" + S[l:]
            l = l - K
        return S    
        

if __name__ == "__main__":
    solu = Solution()
    s = "5F3Z-2e-9-w"
    k = 4
    print(solu.licenseKeyFormatting_a(s, k))